adding premium packs explanation docs (#236)
demisto/content-docs: docs/integrations/preimum_packs.md

---
id: premium_packs
title: Premium Packs setup
---

In this document we will go over the entire process for developing paid content for XSOAR’s marketplace, step by step.

1. Duplicate the demisto/content-external-template repository by clicking on the “Use this template”, then choose to create it as a private repository under your user.

> <img src="../doc_imgs/integrations/demisto/content-external-template.png" width="480"></img>

Reviewer: seems the path is invalid

2. Follow the guidelines mentioned in the template repository in-order to get your pack in there, don't forget to choose your pricing for the pack.
5. Once you have completed the work on the pack you should invite the “xsoar-bot” user as a collaborator to your repository so that we will be able to review your contribution and add it into our build system.
This can be done using this [guide](https://help.github.jp/enterprise/2.11/user/articles/inviting-collaborators-to-a-personal-repository/).
6. Send the details of the pack to the person you are in contact with and he will link your contribution to our system.

> 7. Now the review phase begins, so as a regular review we will leave comments and will ask you to fix them, once we approve we move to the next step.
> 8. When your build passes and we give you the ok you can merge the PR. When you do, we will upload your pack to the XSOAR marketplace, please note that this process may take couple of hours.

Reviewer: ```suggestion
7. Review Phase - we will review your pull request and leave notes, in parallel we have internal an validation process that we will spin up to validate your pack.
In order to approve your pull request, we need:
- Pull request build must pass
- Fix all our notes in the pull request
- Internal validation process must pass - we will update you if it will fail and why.
8. Only after we approved your pull request, you are should merge pull request.
```

---
id: premium_packs
title: Premium Packs setup
---

> First of all, we would like to thank you for trusting us with your content. We hope to make this process as smooth and easy for you as possible.

Reviewer: remove this

In this document we will go over the entire process for developing paid content for XSOAR’s marketplace, step by step.

> If anything is confusing or unclear,  please feel free to reach out - we would love to hear your feedback and improve this documentation with it.

Reviewer: remove this

> 1. Duplicate the demisto/content-external-template repository by clicking on the “Use this template”, then choose to create it as a private repository under your user.

Reviewer 2: can you add a screenshot?

2. Setup your development environment using [this](https://xsoar.pan.dev/docs/integrations/dev-setup).
3. Create your new pack under the Packs folder in the repository and use the documentation we have under the following [guide](https://xsoar.pan.dev/docs/integrations/getting-started-guide) to get you started.
4. Under your pack you will need to update your pack_metadata.json file using the description provided in [the following guide](https://xsoar.pan.dev/docs/integrations/packs-format#pack_metadatajson).
5. Once you have completed the work on the pack you should invite the “xsoar-bot” user as a collaborator to your repository so that we will be able to review your contribution and add it into our build system.
This can be done using this [guide](https://help.github.jp/enterprise/2.11/user/articles/inviting-collaborators-to-a-personal-repository/).

> 6. Send the details of the pack to the person you are in contact with and he will link your contribution to our system.

Reviewer 3: he needs to be invited as a collaborator too, both to review and to create the json file we need for the private repo

Author: the xsoar-bot should be enough

7. Now the review phase begins, so as a regular review we will leave comments and will ask you to fix the comments we leave there, once we approve we move to the next step.

> 8. When your build passes and we give you the ok you can merge the PR. When you do, our build will pull your changes once again and upload your pack to the XSOAR marketplace.

Reviewer: lets mention here that merging the PR in their repo still doesn't mean that the pack will be available in the marketplace. for that we should trigger a process on our side that will take their code, pack it and push it to marketplace

> 8. When your build passes and we give you the ok you can merge the PR, which will result in our build to pull your changes once again and upload your pack to the marketplace of XSOAR.

Reviewer 3: ```suggestion
8. When your build passes and we give you the ok you can merge the PR. When you do our build will pull your changes once again and upload your pack to the XSOAR marketplace.
```

> 8. When your build passes and we give you the ok you can merge the PR. When you do, we will trigger a build manualy which will pull your changes once again and will upload your pack to the XSOAR marketplace.

Reviewer: No need to explain the internals of what we will happen on our side.

---
id: premium_packs
title: Premium Packs setup
---

In this document we will go over the entire process for developing paid content for XSOAR’s marketplace, step by step.

1. Duplicate the demisto/content-external-template repository by clicking on the “Use this template”, then choose to create it as a private repository under your user.  

> <img src="../doc_imgs/integrations/demisto:content-external-template.png" width="480"></img>  

Reviewer: The image seems very small. And it is not clickable. Increase the image size and see how/can we make it clickable (in order to see full image)

Author: Updated the size, images are not clickable in general, you can right-click on them and open in a new tab
